### DISCUSSION TITLE
Fix missing quota information

### DISCUSSION THREAD

**Discussion by bdickensheets** [abraunegg/onedrive#1539] [Q&A]
I am hoping that someone with more knowledge of Microsoft OneDrive/SharePoint/Azure can suggest how to fix the warning generated when SharePoint does not return quota information:
```
WARNING: OneDrive quota information is being restricted or providing a zero value. Please fix by speaking to your OneDrive / Office 365 Administrator.
```

This is essentially a follow-up to issue #1225 , for someone who is an administrator of their M365 ecosystem. I simply don't know where to look for the setting that controls whether or not quota information is returned to the OneDrive client.

There are no obvious settings in the SharePoint admin portal, and my familiarity with the Azure admin portal is cursory at best.

Google searches have been fruitless (potentially because I don't know what to search for).

If someone can point me in the right direction, I'd be happy to write up an FAQ entry that could hopefully help others with the same issue in the future.

**Comment by abraunegg** [comment]
@bdickensheets 
Whilst I cannot advise here, this message gets displayed when the 'remaining' value in the JSON response is '0', when it technically should be returning some sort of value.

This data comes from the API call to query the drive we are configured to use (default drive, configured alternate drive ID or a SharePoint Drive).

A normal response from the API will include the following data:
```
	"quota": {
		"deleted": 365724,
		"remaining": 1099511627776,
		"state": "normal",
		"total": 1099511627776,
		"used": 5111341
	},
```

The above response comes from a Office365 Business Tenant used to test this client when developing it. It has the 'default' Microsoft configuration with zero changes - and the quota remaining data is provided in this state.

When querying a configured SharePoint site:
```
	"quota": {
		"deleted": 46785,
		"remaining": 27487767932430,
		"state": "normal",
		"total": 27487790694400,
		"used": 47873899
	},
```
Again, using default Microsoft configuration, the required data is returned.

When that particular message is presented, when using business accounts - the 'remaining' value returned is '0' - meaning that OneDrive, via the API call, cannot advise you how much storage you have left.

I would advise you to open a case with Microsoft to get your answer, it is highly doubtful anyone who is watching | reading things here would be able to assist further - happy to be proven wrong on that point however.

**Reply by bdickensheets** [reply]
@abraunegg Appreciate the time for a response and the additional context for the message. It's an excellent suggestion to check with Microsoft--I'll do that and report back here when I get an answer sorted out.
